web/src/engine/websites/LineWebtoon.ts

```diff
@@ -0,0 +1,261 @@
+    }
+
+    private async GetChaptersFromPage(manga: Manga, page: number): Promise<Chapter[]> {
+        const data = await FetchCSS<HTMLAnchorElement>(new Request(new URL(`${manga.Identifier}&page=${page}`, this.URI)), this.queryChapters);
+        return data.map(element => {
+            const { id, title } = ChapterExtractor.call(this, element);
+            return new Chapter(this, manga, id, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        const data = await FetchWindowScript(new Request(new URL(chapter.Identifier, this.URI)), this.pageScript, 1500);
+        if (!Array.isArray(data)) return [];
+        return typeof data[0] === 'string' ? (data as Array<string>).map(page => new Page(this, chapter, new URL(page))) : this.CreatePagesfromData(chapter, data as PageData[]);
+    }
+
+    private CreatePagesfromData(chapter: Chapter, data: PageData[]): Page<PageData>[] {
+        return data.map(page => new Page<PageData>(this, chapter, new URL(this.URI), { Referer: this.URI.origin, ...page }));
+    }
+
+    public override async FetchImage(page: Page<PageData>, priority: Priority, signal: AbortSignal): Promise<Blob> {
+        if (!page.Parameters?.width) return await Common.FetchImageAjax.call(this, page, priority, signal, true);
+        const payload = page.Parameters;
+        return this.imageTaskPool.Add(async () => {
+            return DeScramble(new ImageData(payload.width, payload.height,), async (_, ctx) => {
+                ctx.canvas.width = payload.width;
+                ctx.canvas.height = payload.height;
+
+                if (payload.background.image) {
+                    const image = await LoadImage(payload.background.image);
+                    ctx.canvas.width = image.width;
+                    ctx.canvas.height = image.height;
+                    ctx.drawImage(image, 0, 0);
+                }
+                if (payload.background.color) {
+                    ctx.fillStyle = payload.background.color;
+                    ctx.fillRect(0, 0, ctx.canvas.width, ctx.canvas.height);
+                }
+                for (const layer of payload.layers) {
+                    const type = layer.type.split('|');
+                    if (type[0] === 'image') {
+                        const image = await LoadImage(layer.asset);
+                        if (type[1] === 'text') {
+                            AdjustTextLayerVisibility(layer, image, ctx.canvas);
+                        }
+                        // TODO: process layer.keyframes in case top/left/width/height is animated?
+                        ctx.drawImage(image, layer.left, layer.top, layer.width || image.width, layer.height || image.height);
+                    }
+                }
+            });
+        }, priority, signal);
+    }
+}
+
+function AdjustTextLayerVisibility(layer: ImageLayer, textLayer: HTMLImageElement, canvas: OffscreenCanvas) {
```

Comment:
The method only uses `textWidth` and `textHeight` therefore i would suggest to only inject these two properties into the method instead of the whole image element.